src/software/geom/algorithms/step_along_perimeter.cpp

```diff
@@ -0,0 +1,74 @@
+#include "software/geom/algorithms/step_along_perimeter.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "software/geom/algorithms/collinear.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/geom/segment.h"
+
+
+Point stepAlongPerimeter(const Polygon& polygon, const Point& start,
+                         double travelDistance)
+{
+    if (travelDistance == 0.0)
+    {
+        return start;
+    }
+
+
+    std::vector<Segment> polygonSegments = polygon.getSegments();
+    std::size_t startSegmentIndex        = 0;
+
+    // find initial segment which contains start point
+    std::vector<Segment>::iterator it = std::find_if(
+        polygonSegments.begin(), polygonSegments.end(), [&start](const Segment& segment) {
+            return collinear(segment.getStart(), start, segment.getEnd());
+        });
+
+    startSegmentIndex = std::distance(polygonSegments.begin(), it);
+
+
+    std::size_t segmentIndex = startSegmentIndex;
```

Comment:
Empty lines can be removed
```suggestion
    if (travelDistance == 0.0)
    {
        return start;
    }

    std::vector<Segment> polygonSegments = polygon.getSegments();
    std::size_t startSegmentIndex        = 0;

    // find initial segment which contains start point
    std::vector<Segment>::iterator it = std::find_if(
        polygonSegments.begin(), polygonSegments.end(), [&start](const Segment& segment) {
            return collinear(segment.getStart(), start, segment.getEnd());
        });

    startSegmentIndex = std::distance(polygonSegments.begin(), it);

    std::size_t segmentIndex = startSegmentIndex;
```

Reply:
Ok